src/http/request_template.rs

```diff
@@ -103,7 +105,22 @@ impl RequestTemplate {
   /// Sets the body for the request
   fn set_body<C: PathString + HasHeaders>(&self, mut req: reqwest::Request, ctx: &C) -> reqwest::Request {
     if let Some(body) = &self.body {
-      req.body_mut().replace(body.render(ctx).into());
+      match &self.encoding {
+        Encoding::ApplicationJson => {
+          req.body_mut().replace(body.render(ctx).into());
+        }
+        Encoding::ApplicationXWwwFormUrlencoded => {
+          let raw_data: String = body.render(ctx);
+          // The raw_data can be both JSON or plain string. If it is
+          // JSON, we need to parse it, or, we can directly use it.
+          if let Ok(deserialized_data) = serde_json::from_str::<serde_json::Value>(&raw_data) {
+            let form_data = serde_urlencoded::to_string(deserialized_data).unwrap();
+            req.body_mut().replace(form_data.into());
```

Comment:
`ctx` actually has the complete body already, and we are converting that value to string and then string to value and then value to URL encoded. This is going to be very slow and inefficient.